### DISCUSSION TITLE
After connection is `drop()`ed, it becomes parmanently unable to use

### DISCUSSION THREAD

**Discussion by jjangga0214** [bikeshedder/deadpool#279] [Q&A] [answered]
I am using [`diesel_async`](https://github.com/weiznich/diesel_async).

Let's say I have connections in deadpool.
I want 2 connections in total.

```rs
let pool: Pool<AsyncPgConnection> = Pool::builder(deadpool_config)
    .max_size(2) // <== max size is 2
    .build()
    .unwrap();

// share the pool with async thread
let _pool = pool.clone();
tokio::spawn(async move { 
    for i in 0..3 {
        let mut conn = _pool.get().await.unwrap();
        // This works when i == 0 or i == 1, but not when i == 2 
        do_something(&mut conn); 
    }
});
```

So it seems like `drop(conn)` is automatically called at the end of each iteration(which is by Rust Compiler), and the Pool becomes out of connections.

I just want the connection when I need, and then return it back to the pool after use. 
So I want to get it back to the pool for each iteration.

I hoped `drop(conn)` make this happen, as like `Rc` and `Arc`, but apparently it doesn't.
How can I get it to work?

Thanks!

**Comment by bikeshedder** [comment, marked as answer]
I'm wondering about that comment:
```rust
// This works when i == 0 or i == 1, but not when i == 2 
```

The for loop `for i in 0..2 { ... }` will only iterate over `0` and `1`. If you meant to write a loop that goes from `0` to `2` (inclusive) you can use `for i in 0..=2 {...}` instead.

**Reply by jjangga0214** [reply]
Thanks for the reply. It should have been `0..3` or `0..=2`. Just a simple mistake :)

**Reply by bikeshedder** [reply]
So does this fix your problem or is there some issue within `deadpool_diesel`?

**Reply by jjangga0214** [reply]
Actually, the cause was differentl, but it was totally by my subtle mistake. Thank you for the reply anyway!
